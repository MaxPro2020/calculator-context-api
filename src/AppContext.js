import React from 'react';

const AppContext = React.createContext({
  items: [],
});

export default AppContext;
