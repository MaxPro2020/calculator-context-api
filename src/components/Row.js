import React from 'react';
import {
  ListGroup,
  Row,
  Col,
  Form,
  Button,
} from 'react-bootstrap';
import PropTypes from 'prop-types';
import AppContext from '../AppContext';

const RowItem = ({ row, index }) => (
  <AppContext.Consumer>
    {(action) => (
      <ListGroup.Item variant={row.disable ? 'warning' : (row.quantity !== 0 ? 'success' : 'light')}>
        <Row>
          <Col xs={2}>
            <Form.Control
              as="select"
              onChange={(event) => action.updateRow(
                index,
                row,
                row.quantity,
                event.target.value,
                row.disable,
              )}
            >
              <option value="+">+</option>
              <option value="-">-</option>
            </Form.Control>
          </Col>
          <Col xs={5}>
            <Form.Control
              type="number"
              step="0.1"
              value={row.quantity}
              onChange={(event) => action.updateRow(
                index,
                row,
                event.target.value,
                row.operation,
                row.disable,
              )}
            />
          </Col>
          <Col xs={2}>
            <Button
              variant="outline-danger"
              onClick={() => action.removeRow(
                index,
                row,
                row.quantity,
                'delete',
                row.disable,
              )}
            >
              Delete
            </Button>
          </Col>
          <Col xs={3}>
            <Button
              variant={!row.disable ? 'warning' : 'success'}
              onClick={() => action.updateRow(
                index,
                row,
                row.quantity,
                row.operation,
                !row.disable,
              )}
            >
              {!row.disable ? 'Disable' : 'Enable'}
            </Button>
          </Col>
        </Row>
      </ListGroup.Item>
    )}
  </AppContext.Consumer>
);

RowItem.defaultProps = {
  row: {},
};

RowItem.propTypes = {
  row: PropTypes.shape({
    quantity: PropTypes.number,
    operation: PropTypes.string,
    disable: PropTypes.bool,
  }),
  index: PropTypes.number.isRequired,
};

export default RowItem;
