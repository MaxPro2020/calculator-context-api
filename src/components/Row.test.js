import React from 'react';
import renderer from 'react-test-renderer';
import Row from './Row';
import AppContext from '../AppContext';

describe('Row component', () => {
  it('Renders correctly', () => 
  {
    const updateRow = jest.fn();
    const removeRow = jest.fn();
    const index = 1;
    const tree = renderer
      .create(
        <AppContext.Provider
          value={{
            updateRow,
            removeRow
          }}
        >
          <Row index={index} />
        </AppContext.Provider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
