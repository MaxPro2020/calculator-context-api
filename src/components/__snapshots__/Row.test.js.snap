// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row component Renders correctly 1`] = `
<div
  className="list-group-item list-group-item-success"
  disabled={false}
  onClick={[Function]}
>
  <div
    className="row"
  >
    <div
      className="col-2"
    >
      <select
        className="form-control"
        onChange={[Function]}
      >
        <option
          value="+"
        >
          +
        </option>
        <option
          value="-"
        >
          -
        </option>
      </select>
    </div>
    <div
      className="col-5"
    >
      <input
        className="form-control"
        onChange={[Function]}
        step="0.1"
        type="number"
      />
    </div>
    <div
      className="col-2"
    >
      <button
        className="btn btn-outline-danger"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Delete
      </button>
    </div>
    <div
      className="col-3"
    >
      <button
        className="btn btn-warning"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Disable
      </button>
    </div>
  </div>
</div>
`;
